src/lib.rs

```diff
@@ -96,63 +110,48 @@ pub use manifest::{BindingEntry, HostManifest};
 pub use middleware::prometheus;
 
 pub use middleware::Middleware;
-pub use wapc::prelude::WasiParams;
-
-mod actor;
-mod authz;
-mod capability;
-mod dispatch;
-pub mod errors;
-mod extras;
-mod inthost;
-
-#[cfg(feature = "manifest")]
-mod manifest;
-
-pub mod middleware;
-mod plugins;
-mod router;
+pub use wapc::{prelude::WasiParams, WapcHost};
 
 pub type SubjectClaimsPair = (String, Claims<wascap::jwt::Actor>);
 
 use authz::AuthHook;
-use inthost::ACTOR_BINDING;
+use bus::MessageBus;
+use crossbeam::Sender;
 use plugins::PluginManager;
-use router::Router;
 use std::{
     collections::HashMap,
     sync::{Arc, RwLock},
 };
 use wascap::jwt::{Claims, Token};
-use wascc_codec::{
-    capabilities::CapabilityDescriptor,
-    core::{CapabilityConfiguration, OP_BIND_ACTOR},
-    serialize,
-};
+use wascc_codec::{capabilities::CapabilityDescriptor, SYSTEM_ACTOR};
 
 type BindingsList = Vec<(String, String, String)>;
+pub(crate) type RouteKey = (String, String); // (binding, id)
 
 /// Represents an instance of a waSCC host
 #[derive(Clone)]
 pub struct WasccHost {
+    bus: Arc<MessageBus>,
     claims: Arc<RwLock<HashMap<String, Claims<wascap::jwt::Actor>>>>,
-    router: Arc<RwLock<Router>>,
     plugins: Arc<RwLock<PluginManager>>,
     auth_hook: Arc<RwLock<Option<Box<AuthHook>>>>,
     bindings: Arc<RwLock<BindingsList>>,
-    caps: Arc<RwLock<HashMap<router::RouteKey, CapabilityDescriptor>>>,
+    caps: Arc<RwLock<HashMap<RouteKey, CapabilityDescriptor>>>,
     middlewares: Arc<RwLock<Vec<Box<dyn Middleware>>>>,
+    // the key to this field is the subscription subject, and not either a pk or a capid
+    terminators: Arc<RwLock<HashMap<String, Sender<bool>>>>,
     #[cfg(feature = "gantry")]
     gantry_client: Arc<RwLock<Option<gantryclient::Client>>>,
 }
 
 impl WasccHost {
     /// Creates a new waSCC runtime host
-    pub fn new() -> Self {
+    pub fn new() -> WasccHost {
```

Comment:
Why `Self' -> 'WasccHost`?

Reply:
this is leftover from when I was attempting to use generics during a refactor pass. Putting it back to self.